tensorboard/examples/plugins/example_raw_scalars/tensorboard_plugin_example/plugin.py

```diff
@@ -0,0 +1,124 @@
+# Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""A sample plugin to demonstrate reading scalars."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import collections
+import csv
+import json
+import os
+import functools
+import mimetypes
+
+import six
+from werkzeug import wrappers
+import werkzeug
+
+from tensorboard import errors
+from tensorboard import plugin_util
+from tensorboard.backend import http_util
+from tensorboard.data import provider
+from tensorboard.plugins import base_plugin
+from tensorboard_plugin_example import metadata
+from tensorboard.util import tensor_util
+
+scalar_plugin_name = 'scalars'
+plugin_folder = '/data/plugin/example_raw_scalars/'
+
+class ExampleRawScalars(base_plugin.TBPlugin):
+  """Raw summary example plugin for TensorBoard."""
+
+  plugin_name = metadata.PLUGIN_NAME
+
+  def __init__(self, context):
+    """Instantiates ExampleRawScalars.
+
+    Args:
+      context: A base_plugin.TBContext instance.
+    """
+    self._multiplexer = context.multiplexer
+
+  def get_plugin_apps(self):
+    return {
+        '/index.js': self._serve_index,
+        '/scalars': self.scalars_route,
+        '/runInfo': self._serve_run_info,
+        '/static/*': self._serve_file,
+    }
+
+  @wrappers.Request.application
+  def _serve_run_info(self, request):
+    run_tag_mapping = self._multiplexer.PluginRunToTagToContent(
+      scalar_plugin_name
+    )
+
+    # Convert to the form: {runName: [tagName, tagName, ...]}
+    run_info = {}
+    for run in run_tag_mapping:
+      run_info[run] = run_tag_mapping[run].keys()
+
+    return http_util.Respond(request, run_info, 'application/json')
+
+  @wrappers.Request.application
+  def _serve_index(self, request):
+    del request  # unused
+    filepath = os.path.join(os.path.dirname(__file__), "static", "index.js")
+    with open(filepath) as infile:
+      contents = infile.read()
+    return werkzeug.Response(contents, content_type="application/javascript")
+
+  @wrappers.Request.application
+  def _serve_file(self, request):
+    """Returns a resource file."""
+    file_path = request.path
+    if file_path.startswith(plugin_folder):
```

Comment:
In that example,
`/data/plugin/example_raw_scalars/static/foo` > `./static/foo`, which presumably serves file `foo` in the static directory.

`/data/plugin/example_raw_scalars/static/data/plugin/example_raw_scalars/foo` > `./static/data/plugin/example_raw_scalars/foo`, which presumably fails because it doesn't exist in the filesystem.

What I really want is for wildcard path handlers to have an extra param, `wildcard_path_part` or something like that.  Simply modifying [`_route_request()`](https://github.com/tensorflow/tensorboard/blob/master/tensorboard/backend/application.py#L520) didn't work for me, as Werkzeug's `@wrappers.Request.application` only expects `environ` and `start_response`.

When we added support for wildcard handlers in https://github.com/tensorflow/tensorboard/pull/2476/ , it seems we didn't add any usages, and I couldn't find any.  I could use some help to understand the right way to do this!

Reply:
Got it: this is the full request path, not just the path under this
plugin, so it always contains the prefix. The added comment helps.
Thanks.